src/ompl/tools/thunder/src/SPARSdb.cpp

```diff
@@ -967,7 +967,8 @@ bool ompl::geometric::SPARSdb::addStateToRoadmap(const base::PlannerTerminationC
         //this added call to findGraphNeighbors is very inexpensive when granularity is small enough (which it is meant to be).
         findGraphNeighbors(qNew, gnbhd, vnbhd, granularity_);
         // if roadmap has only one connected component when we try to add a node, skip the expensive attempt to add it as a connectivity node.
-        if (getNumConnectedComponents() == 1 && vnbhd.size()) {
+        const bool check_node_cc {check_connected_components_ && getNumConnectedComponents() > 1};
+        if (!check_node_cc && vnbhd.size()) {
```

Comment:
Why isn't this all in the if?

Reply:
What is `vnbhd`? Can keep the if clause separate if it is more meaningful. A better name for `check_node_cc` can be `check_if_connect_the_disconnected_components`. Verbose but clear.